sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -291,10 +308,10 @@ private[orc] case class OrcTableScan(
     val orcFormat = new DefaultSource
     val dataSchema =
       orcFormat
-          .inferSchema(sqlContext, Map.empty, inputPaths)
-          .getOrElse(sys.error("Failed to read schema from target ORC files."))
+        .inferSchema(sqlContext, Map.empty, inputPaths)
+        .getOrElse(sys.error("Failed to read schema from target ORC files."))
     // Sets requested columns
-    addColumnIds(dataSchema, attributes, conf)
+    OrcRelation.setRequiredColumns(conf, dataSchema, StructType.fromAttributes(attributes))
```

Comment:
The original method is in `OrcFileScan`, which is a case class, and is hard to reuse. This PR refactors `addColumnIds` and `fillObject` in `OrcFileScan` and moves them to `object OrcRelation` so that they can be easily reused.